dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/service/TestDataService.java

```diff
@@ -0,0 +1,168 @@
+            super(message);
+        }
+    }
+
+  /**
+   * Create test data required for testing the FDC Delayed Pickup logic
+   * (	 * @see <a href="https://dsdmoj.atlassian.net/browse/DCES-356">DCES-365</a> for test specification.)
+   * @param testType: One of the test types defined in enum FdcTestType
+   * @param recordsToUpdate Number of records to update
+   * @return A set of FDC IDs updated as required
+   */
+  public Set<Integer> createDelayedPickupTestData(FdcTestType testType, int recordsToUpdate){
+    Set<Integer> repOrderIds = getFdcDelayCandidateRepOrderIds(5, LocalDate.of(2015,1,1), recordsToUpdate);
+    Set<Integer> fdcIds = new HashSet<>();
+    if (repOrderIds != null && !repOrderIds.isEmpty()) {
+      repOrderIds.forEach(repOrderId -> {
+        FdcContribution fdcContribution = createFdcContribution(repOrderId, "Y", "Y", null, WAITING_ITEMS);
+        int fdcId = fdcContribution.getId();
+        fdcIds.add(fdcId);
+        FdcItem fdcItem = FdcItem.builder()
+                .fdcId(fdcId)
+                .userCreated("DCES")
+                .dateCreated(LocalDate.now())
+                .build();
+        createFdcItem(fdcItem);
+        createAdditionalNegativeTypeTestData(testType, repOrderId, fdcId);
+      });
+    } else {
+      throw new TestDataServiceException("No candidate rep orders found for delayed pickup test type " + testType);
+    }
+    return fdcIds;
+  }
+
+  /**
+   * Create test data required for testing the FDC Fast Track  logic
+   * (	 * @see <a href="https://dsdmoj.atlassian.net/browse/DCES-356">DCES-367</a>,
+   * <a href="https://dsdmoj.atlassian.net/browse/DCES-356">DCES-377</a> and
+   * <a href="https://dsdmoj.atlassian.net/browse/DCES-356">DCES-378</a>for test specification (this one method caters to all of them).)
+   * @param fdcAccelerationType: One of the acceleration types defined in enum FdcAccelerationType
+   * @param testType: One of the test types defined in enum FdcTestType
+   * @param recordsToUpdate Number of records to update
+   * @return A set of FDC IDs updated as required
+   */
+
+  public Set<Integer> createFastTrackTestData( FdcAccelerationType fdcAccelerationType, FdcTestType testType, int recordsToUpdate){
+    Set<Integer> repOrderIds = getFdcFastTrackCandidateRepOrderIds(-3, LocalDate.of(2015,1,1), recordsToUpdate);
+    Set<Integer> fdcIds = new HashSet<>();
+    if(repOrderIds == null || repOrderIds.isEmpty() ){
+      throw new TestDataServiceException("No candidate rep orders found for fast-track pickup test type " + testType);
+    }
+    repOrderIds.forEach(repOrderId -> {
+      maatApiClient.updateRepOrder(UpdateRepOrder.builder().repId(repOrderId).sentenceOrderDate(LocalDate.now().minusMonths(3)).build());
+      String manualAcceleration = (fdcAccelerationType == FdcAccelerationType.POSITIVE)?"Y":null;
+      int fdcId = createFdcContribution(repOrderId, "Y", "Y", manualAcceleration, WAITING_ITEMS).getId();
+      fdcIds.add(fdcId);
+
```

Comment:
These lines from the original class (FdcTestDataCreatorService) seem to have been removed:

                if (fdcAccelerationType.equals(FdcAccelerationType.PREVIOUS_FDC)) {
                    testDataClient.createFdcContribution(new CreateFdcContributionRequest(repOrderId, "Y", "Y", null, SENT));
                }

Reply:
These have been merged into a condition below that line:

`      if (fdcAccelerationType.equals(FdcAccelerationType.PREVIOUS_FDC)) {
        createFdcContribution(repOrderId, "Y", "Y", null, SENT);
        if(testType.equals(FdcTestType.NEGATIVE_PREVIOUS_FDC)){
          fdcItemBuilder = fdcItemBuilder.adjustmentReason("Other");
        }
      }`